src/approaches/oracle_approach.py

```diff
@@ -93,96 +88,167 @@ def _get_options_by_names(env_name: str,
 
+    # Objects
+    block = Variable("?block", block_type)
+    robot = Variable("?robot", robot_type)
+    target = Variable("?target", target_type)
+
+    # Predicates
     IsBlock, IsTarget, Covers, HandEmpty, Holding = \
         _get_predicates_by_names(CFG.env, ["IsBlock", "IsTarget", "Covers",
                                            "HandEmpty", "Holding"])
 
-    if options_are_typed:
-        Pick, Place = _get_options_by_names(CFG.env, ["Pick", "Place"])
-    else:
+    # Options
+    if CFG.env in ("cover", "cover_hierarchical_types"):
         PickPlace, = _get_options_by_names(CFG.env, ["PickPlace"])
+    elif CFG.env in ("cover_typed_options", "cover_multistep_options"):
+        Pick, Place = _get_options_by_names(CFG.env, ["Pick", "Place"])
+    if CFG.env == "cover_multistep_options" and \
+        CFG.cover_multistep_use_learned_equivalents:
+        LearnedEquivalentPick, LearnedEquivalentPlace = _get_options_by_names(
+            CFG.env, ["LearnedEquivalentPick", "LearnedEquivalentPlace"])
 
     nsrts = set()
 
     # Pick
-    block = Variable("?block", block_type)
-    robot = Variable("?robot", robot_type)
-    parameters = [block, robot] if include_robot_in_holding else [block]
-    if options_are_typed:
-        option_vars = [block]
-        option = Pick
-    else:
-        option_vars = []
-        option = PickPlace
+    parameters = [block]
+    holding_predicate_args = [block]
+    if CFG.env == "cover_multistep_options":
+        parameters.append(robot)
+        holding_predicate_args.append(robot)
     preconditions = {LiftedAtom(IsBlock, [block]), LiftedAtom(HandEmpty, [])}
-    if include_robot_in_holding:
-        add_effects = {LiftedAtom(Holding, [block, robot])}
-    else:
-        add_effects = {LiftedAtom(Holding, [block])}
+    add_effects = {LiftedAtom(Holding, holding_predicate_args)}
     delete_effects = {LiftedAtom(HandEmpty, [])}
-    def pick_sampler(state: State, rng: np.random.Generator,
-                     objs: Sequence[Object]) -> Array:
-        assert len(objs) == 2 if include_robot_in_holding else len(objs) == 1
-        b = objs[0]
-        assert b.is_instance(block_type)
-        if options_are_typed:
-            lb = float(-state.get(b, "width")/2)  # relative positioning only
-            ub = float(state.get(b, "width")/2)  # relative positioning only
-        else:
-            lb = float(state.get(b, "pose") - state.get(b, "width")/2)
-            lb = max(lb, 0.0)
-            ub = float(state.get(b, "pose") + state.get(b, "width")/2)
-            ub = min(ub, 1.0)
-        return np.array(rng.uniform(lb, ub, size=(1,)), dtype=np.float32)
+
+    if CFG.env in ("cover", "cover_hierarchical_types"):
+        option = PickPlace
+        option_vars = []
+    elif CFG.env in ("cover_typed_options", "cover_multistep_options"):
+        option = Pick
+        option_vars = [block]
+    if CFG.env == "cover_multistep_options" and \
```

Comment:
minor, can you reorder these so that we don't end up creating and then immediately overriding `option` and `option_vars`? one possibility is to move this case up before the previous one, so:

```
if CFG.env in ("cover", "cover_hierarchical_types"):
...
elif CFG.env == "cover_multistep_options" and CFG.cover_multistep_use_learned_equivalents:
...
else:
    assert CFG.env in ("cover_typed_options", "cover_multistep_options")
```

this comment also applies to a similar pattern with Place